design/design-doc-stacks.md

```diff
@@ -301,24 +386,25 @@ Possible solutions for package processing are listed below, along with their pro
   * Package images only contain the package contents, no processing logic is included in the package
   * The SM starts a job pod with an init container for the package image and copies all its contents to a shared volume in the job pod.  The Crossplane package processing logic runs in the main pod container and runs over the shared volume, sending all artifacts to `stdout`.
   * PRO: Package images are significantly smaller since they will only contain yaml files and icons, no binaries or libraries.
+  * PRO: The processing/unpacking logic can easily move to its own image that can be used as a base layer in the future (or a CLI tool), so this approach is not very divergent.
   * CON: The SM needs to understand package content and layout.
 * CLI tool
   * The unpacking logic could be built into a CLI tool that can be run over a package image.
   * PRO: This is the most versatile option as it can be used in contexts outside of a Crossplane cluster.
   * CON: Doesn't integrate very cleanly into the Crossplane mainline scenario with the SM.
 
-**Recommendation:** It is recommended to start with the "Job initContainer and shared volume" approach that copies the package contents from the package initContainer to a shared volume and has a Crossplane container main entry point perform the processing logic over the shared volume contents, sending out the artifacts to `stdout`.
-Stack packages should only have a few KB of yaml files, so it's a significant concern to blow that image size up to many MB right now.
-The processing/unpacking logic can easily move to its own image that can be used as a base layer in the future (or a CLI tool), so this approach is not very divergent.
-It is a good place to start and iterate over as we learn more without investing much effort that cannot be reused.
+Each of these designs offered a good place to start.  Through iteration over time we will learn more, hopefully without investing much effort that cannot be reused.
 
 ## Questions and Open Issues
```

Comment:
In the `Terminology` section, would it make sense for us to split out `Stack Manager` from `Stack lifecycle manager`? I could see a future in which install/uninstall are managed by a different component than upgrades. For example, maybe we decide to have install and uninstall triggered by an operator outside of the cluster, but decide to have upgrades triggered automatically by something inside the cluster.

Reply:
I do think there are some utility and efficiency reasons to support out-of-cluster tooling, including:
- cleaning up stale stacks after the stack manager has itself become broken or removed
- reducing the footprint of a stack deployed system

But, I think the current thinking is to treat the Stack Manager as a full life-cycle governor. While there may be additional ways to install and remove a Stack, I think we can avoid theorizing about their benefits at this time until we are ready to act on those ideas (and revise the design docs to match).